mysk-lib/src/models/elective_subject/db.rs

```diff
@@ -242,31 +236,35 @@ impl QueryDb<QueryableElectiveSubject, SortableElectiveSubject> for DbElectiveSu
         }
 
         for (i, section) in where_sections.iter().enumerate() {
-            // add WHERE or AND before each section
-            query.push(if i == 0 { " WHERE " } else { "AND " });
-            // len of sql should be len of params + 1
-            // loop through index of sql
-            //   push sql[i]
-            //   if i < len of params
-            //     push params[i]
+            query_builder.push(if i == 0 { " WHERE " } else { " AND " });
             for (j, sql) in section.sql.iter().enumerate() {
-                query.push(sql);
+                query_builder.push(sql);
                 if j < section.params.len() {
-                    match &section.params[j] {
-                        QueryParam::Int(v) => query.push_bind(v),
-                        QueryParam::Float(v) => query.push_bind(v),
-                        QueryParam::String(v) => query.push_bind(v),
-                        QueryParam::Bool(v) => query.push_bind(v),
-                        QueryParam::Uuid(v) => query.push_bind(v),
-                        QueryParam::ArrayInt(v) => query.push_bind(v),
-                        QueryParam::ArrayFloat(v) => query.push_bind(v),
-                        QueryParam::ArrayString(v) => query.push_bind(v),
-                        QueryParam::ArrayBool(v) => query.push_bind(v),
-                        QueryParam::ArrayUuid(v) => query.push_bind(v),
+                    match section.params.get(j) {
+                        Some(QueryParam::Int(v)) => query_builder.push_bind(*v),
+                        Some(QueryParam::Float(v)) => query_builder.push_bind(*v),
+                        Some(QueryParam::Uuid(v)) => query_builder.push_bind(*v),
+                        Some(QueryParam::String(v)) => query_builder.push_bind(v.clone()),
+                        Some(QueryParam::ArrayInt(v)) => query_builder.push_bind(v.clone()),
+                        Some(QueryParam::ArrayUuid(v)) => query_builder.push_bind(v.clone()),
+                        _ => unreachable!(),
```

Comment:
I wouldn't do this as it would be very hard to debug in the future if we were to add the other currently unreachable types